cmake/compilerFlags.cmake

```diff
@@ -20,17 +20,24 @@ if ( MINGW OR UNIX OR MSYS ) # MINGW, Linux, APPLE, CYGWIN
         endif()
     endif()
 
-    if (COMPILER_IS_GCC OR COMPILER_IS_CLANG)
+    if (COMPILER_IS_GCC)
+        add_compile_options(-Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS)
+        if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 8.0 )
+            add_compile_options(-fstack-clash-protection -fcf-protection)

```

Comment:
clang also provides stack protection and gcc < 8 has `-fstack-protection=strong`.

Reply:
I got errors with GCC 5.4 and I think I added that if condition after googling a bit:

https://travis-ci.org/piponazo/exiv2/jobs/564650878

However, I could not find good references telling me from which version the compiler flags are available. Do you know about some resource with such information ?

Reply:
I just noticed that `-fstack-protection=strong` is used for all the versions in line 38. These are other flags.